adafruit_displayio_layout/widgets/cartesian.py

```diff
@@ -457,54 +457,162 @@ def _draw_pointers(self, x: int, y: int) -> None:
 
         self.append(self._pointer)
 
-    def update_pointer(self, x: int, y: int) -> None:
-        """updater_pointer function
-        helper function to update pointer in the plane
+    def _calc_local_xy(self, x: int, y: int) -> (int, int):
+        local_x = (
+            int((x - self._xrange[0]) * self._factorx * self._valuex) + self._nudge_x
+        )
+        # details on `+ (self.height - 1)` :
+        # the bitmap is set to self.width & self.height
+        # but we are only allowed to draw to pixels 0..height-1 and 0..width-1
+        local_y = (
+            int((self._yrange[0] - y) * self._factory * self._valuey)
+            + (self.height - 1)
+            + self._nudge_y
+        )
+        return (local_x, local_y)
+
+    def _check_local_x_in_range(self, local_x):
+        return 0 <= local_x < self.width
+
+    def _check_local_y_in_range(self, local_y):
+        return 0 <= local_y < self.height
+
+    def _check_local_xy_in_range(self, local_x, local_y):
+        return self._check_local_x_in_range(local_x) and self._check_local_y_in_range(
+            local_y
+        )
+
+    def _check_x_in_range(self, x):
+        return self._xrange[0] <= x <= self._xrange[1]
+
+    def _check_y_in_range(self, y):
+        return self._yrange[0] <= y <= self._yrange[1]
+
+    def _check_xy_in_range(self, x, y):
+        return self._check_x_in_range(x) and self._check_y_in_range(y)
+
+    def _add_point(self, x: int, y: int) -> None:
+        """_add_point function
+        helper function to add a point to the graph in the plane
         :param int x: ``x`` coordinate in the local plane
         :param int y: ``y`` coordinate in the local plane
         :return: None
         rtype: None
         """
-        local_x = int((x - self._xrange[0]) * self._factorx) + self._nudge_x
-        local_y = (
-            int((self._yrange[0] - y) * self._factory) + self.height + self._nudge_y
+        print("")
+        print(
```

Comment:
lets remove the print statements, or add a `verbose` argument used to enable them.

Reply:
yeah - sorry ;-)
i added an verbose flag